src/api/ApiCall.hh

```diff
@@ -0,0 +1,143 @@
+#ifndef RA_API_CALL_HH
+#define RA_API_CALL_HH
+#pragma once
+
+#include "services\IThreadPool.hh"
+#include "services\ServiceLocator.hh"
+
+#include <string>
+
+namespace ra {
+namespace api {
+
+enum class ApiResult
+{    
+    None = 0,       // unspecified
+    Success,        // call was successful
+    Error,          // an error occurred
+    Failed,         // call was unsuccessful, but didn't error
+    Unsupported,    // call is not supported and was not made
+};
+
+struct ApiRequestBase
+{
+protected:
+    ApiRequestBase() noexcept = default;
+    ~ApiRequestBase() noexcept = default;
+    ApiRequestBase(const ApiRequestBase&) = default;
+    ApiRequestBase& operator=(const ApiRequestBase&) = default;
+    ApiRequestBase(ApiRequestBase&&) noexcept = default;
+    ApiRequestBase& operator=(ApiRequestBase&&) noexcept = default;
```

Comment:
No, if the forced compiler generated constructor is `noexcept` and/or `constexpr` then the implicitly generated ones are identical.

The `define or delete them all` rule is if you do do `=delete` or make a constructor/assignment user-defined instead of compiler generated (has a body instead of `=default`).